File: src/main/java/com/shiminfxcvii/turing/model/dto/DictDTO.java
package com.shiminfxcvii.turing.model.dto;

import com.shiminfxcvii.turing.model.query.PageQuery;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

/**
 * 字典 DTO
 *
 * @author ShiminFXCVII
 * @since 12/30/2022 4:03 PM
 */
@Getter
@Setter
public class DictDTO extends PageQuery {
    /**
     * 字典 id
     */
    private String id;
    /**
     * 字典类型
     */
    @NotBlank(message = "字典类型不能为空")
    private String type;
    /**
     * 上级字典 id
     */
    private String pid;
    /**
     * 字典名称
     */
    @NotBlank(message = "字典名称不能为空")
    private String name;
    /**
     * 字典值
     */
    @NotBlank(message = "字典值不能为空")
    private String value;
    /**
     * 字典说明
     */
    private String description;
    /**
     * 字典排序
     */
    private Integer sort;
    /**
     * 下级字典
     */
    private List<DictDTO> children;
}

File: turing-resource-server/src/main/java/com/simonvonxcvii/turing/resource/server/model/dto/DictDTO.kt
package com.simonvonxcvii.turing.resource.server.model.dto

import com.simonvonxcvii.turing.resource.server.model.query.PageQuery
import jakarta.validation.constraints.NotBlank
import java.io.Serializable

/**
 * DTO for [com.simonvonxcvii.turing.resource.server.entity.Dict]
 * 
 * @author Simon Von
 * @since 12/30/2022 4:03 PM
 */
data class DictDTO(
    /**
     * 字典 id
     */
    var id: Int? = null,

    /**
     * 字典类型
     */
    @field:NotBlank(message = "字典类型不能为空")
    var type: String? = null,

    /**
     * 上级字典 id
     */
    var pid: Int? = null,

    /**
     * 字典名称
     */
    @field:NotBlank(message = "字典名称不能为空")
    var name: String? = null,

    /**
     * 字典值
     */
    @field:NotBlank(message = "字典值不能为空")
    var value: String? = null,

    /**
     * 字典说明
     */
    var description: String? = null,

    /**
     * 字典排序
     */
    var sort: Int? = null,

    /**
     * 下级字典
     */
    var children: MutableList<DictDTO>? = mutableListOf()
) : Serializable, PageQuery()
